### DISCUSSION TITLE
Bun v1.0.2

### DISCUSSION THREAD

**Discussion by Jarred-Sumner** [oven-sh/bun#5482] [Announcements]
To install Bun v1.0.2:
```bash
curl https://bun.sh/install | bash
# or you can use npm
# npm install -g bun
```

To upgrade to Bun v1.0.2:
```bash
bun upgrade
```

**[Read Bun v1.0.2's release notes on Bun's blog](https://bun.sh/blog/bun-v1.0.2)**

## What's Changed
* fix(bun-lambda) Fix API Gateway V1 events and expand on Lambda documentation by @mkossoris in https://github.com/oven-sh/bun/pull/5161
* fix lifecycle docu by @ximex in https://github.com/oven-sh/bun/pull/5159
* docs: fix typos by @s-rigaud in https://github.com/oven-sh/bun/pull/5151
* Updated Lambda readme by @tsndr in https://github.com/oven-sh/bun/pull/5127
* Add missing full stop on nodejs-apis.md by @diogo405 in https://github.com/oven-sh/bun/pull/5072
* update dev build instruction for arch by @mi4uu in https://github.com/oven-sh/bun/pull/5069
* fix typo and grammar errors in bunfig.toml by @xNaCly in https://github.com/oven-sh/bun/pull/5046
* Update simple.md by @tomredman in https://github.com/oven-sh/bun/pull/4997
* Update hot.md by @nazeelashraf in https://github.com/oven-sh/bun/pull/4990
* docs: fix typo in import.meta.resolve by @jonathantneal in https://github.com/oven-sh/bun/pull/5146
* [Docs] Use git's `--global` flag for lockfile diffs instead of manually editing config files. by @Southpaw1496 in https://github.com/oven-sh/bun/pull/5143
* Various docs by @colinhacks in https://github.com/oven-sh/bun/pull/5201
* docs: Made bun-types install as dev dependency in example by @MasterGordon in https://github.com/oven-sh/bun/pull/5120
* js/node/stream.js: call write() callback when encoding is not provided by @cfal in https://github.com/oven-sh/bun/pull/4841
* Correct the configuration file names. by @nathanhammond in https://github.com/oven-sh/bun/pull/5234
* decode regex if needed by @dylan-conway in https://github.com/oven-sh/bun/pull/5167
* Update misleading documentation link by @0x346e3730 in https://github.com/oven-sh/bun/pull/5227
* file.exists() needs to be awaited to get the value by @amt8u in https://github.com/oven-sh/bun/pull/5061
* docs(runtime): fix jsx FragmentFactory output example by @zongzi531 in https://github.com/oven-sh/bun/pull/5243
* Add informative message on `bun create react` by @colinhacks in https://github.com/oven-sh/bun/pull/5248
* chore: make comment grammatically correct by @G-Rath in https://github.com/oven-sh/bun/pull/5140
* docs(runtime): fix plugins loader extensions typo by @zongzi531 in https://github.com/oven-sh/bun/pull/5250
* avoid inserting extraneous"accept-encoding" header by @iidebyo in https://github.com/oven-sh/bun/pull/5057
* fix(node/fetch): Make data URL fetch consistent with node by @davidmhewitt in https://github.com/oven-sh/bun/pull/5126
* docs: update lockfile diff instructions by @gtramontina in https://github.com/oven-sh/bun/pull/5275
* add uninstall instructions by @browner12 in https://github.com/oven-sh/bun/pull/5311
* docs(guide): fix expect assertion example in guide for `spyOn` by @winghouchan in https://github.com/oven-sh/bun/pull/5294
* chore(docs): include missing links to Node.js APIs by @styfle in https://github.com/oven-sh/bun/pull/5281
* Fixed api & cli docs typo. by @jamesgordo in https://github.com/oven-sh/bun/pull/5262
* fix(runtime): require cache should not include unevaluated ESM modules. by @paperdave in https://github.com/oven-sh/bun/pull/5233
* Make --watch instant by @Jarred-Sumner in https://github.com/oven-sh/bun/pull/5236
* feat(nodejs): implement `os.availableParallelism` by @WingLim in https://github.com/oven-sh/bun/pull/5109
* fix(console.log) fix printing long custom format by @cirospaciari in https://github.com/oven-sh/bun/pull/5164
* fix(node:dns): fix crash and compatibility issues with `resolve` by @Hanaasagi in https://github.com/oven-sh/bun/pull/5200
* fix(doc): Add "compilerOptions" to bun-types README.md by @philolo1 in https://github.com/oven-sh/bun/pull/5325
* fix(BunFile.slice) fix slice when length is greater than the size by @cirospaciari in https://github.com/oven-sh/bun/pull/5186
* More docs & helptext cleanup by @colinhacks in https://github.com/oven-sh/bun/pull/5229
* doc(guides): update sveltekit guide  by @mroyme in https://github.com/oven-sh/bun/pull/5285
* modules documentation didn't have correct import example by @miccou in https://github.com/oven-sh/bun/pull/5225
* fix link to "local template" by @desm in https://github.com/oven-sh/bun/pull/5115
* chore: test for overwriting _resolveFilename by @paperdave in https://github.com/oven-sh/bun/pull/5244
* Fix bug with multiline string in CRLF terminated files (#4893) by @tikotzky in https://github.com/oven-sh/bun/pull/5318
* fix(runtime): make most globals configurable/deletable, allow resuming the console iterator by @paperdave in https://github.com/oven-sh/bun/pull/5216
* fix(Bun.serve) fix buffering edge case by @cirospaciari in https://github.com/oven-sh/bun/pull/5152
* Update nextjs.md by @kryparnold in https://github.com/oven-sh/bun/pull/4905
* Update simple.md by @TwanLuttik in https://github.com/oven-sh/bun/pull/4881
* Update nuxt.md by @s0h311 in https://github.com/oven-sh/bun/pull/4452
* Remove the ability to configure lockfile location. by @nathanhammond in https://github.com/oven-sh/bun/pull/5335
* In `bunx`, always get latest version when @latest is explicitly passed by @Jarred-Sumner in https://github.com/oven-sh/bun/pull/5346
* Fix typo in HTTPThread name by @chrisbodhi in https://github.com/oven-sh/bun/pull/5376
* docs - Add "workspace:*" to workspace docs. by @dylang in https://github.com/oven-sh/bun/pull/5379
* fix(docs): Fix the text that `bun run --bun` is the same as `bun` by @DuGlaser in https://github.com/oven-sh/bun/pull/4647
* fix warnings during bun run publish-layer by @nangchan in https://github.com/oven-sh/bun/pull/5419
* fix(runtime): emit `node:net` connect error event vs throw by @paperdave in https://github.com/oven-sh/bun/pull/5336
* v8 date parser tests by @dylan-conway in https://github.com/oven-sh/bun/pull/5332
* async-ify all node:fs functions by @Jarred-Sumner in https://github.com/oven-sh/bun/pull/5360
* fix dockerfile by @Jarred-Sumner in https://github.com/oven-sh/bun/pull/5439
* fix http set cookie headers by @dylan-conway in https://github.com/oven-sh/bun/pull/5428
* fix(nitro) fix sourcemaps and JSSink closing by @cirospaciari in https://github.com/oven-sh/bun/pull/5422
* Update docs/quickstart.md by @sonyarianto in https://github.com/oven-sh/bun/pull/5425
* dup and close file descriptors by @dylan-conway in https://github.com/oven-sh/bun/pull/5341
* Make `bun run --silent` omit `"error: "..." exited with code 1`  by @Jarred-Sumner in https://github.com/oven-sh/bun/pull/5459
* Does not fix #4622 by @Jarred-Sumner in https://github.com/oven-sh/bun/pull/5452
* fix(proxy): allow empty string `http_proxy` env by @Hanaasagi in https://github.com/oven-sh/bun/pull/5464
* Implement `URL.canParse` by @Jarred-Sumner in https://github.com/oven-sh/bun/pull/5463
* Fixes #5461 by @Jarred-Sumner in https://github.com/oven-sh/bun/pull/5467
* Fixes #5465 by @Jarred-Sumner in https://github.com/oven-sh/bun/pull/5468

## New Contributors
* @mkossoris made their first contribution in https://github.com/oven-sh/bun/pull/5161
* @ximex made their first contribution in https://github.com/oven-sh/bun/pull/5159
* @s-rigaud made their first contribution in https://github.com/oven-sh/bun/pull/5151
* @tsndr made their first contribution in https://github.com/oven-sh/bun/pull/5127
* @mi4uu made their first contribution in https://github.com/oven-sh/bun/pull/5069
* @xNaCly made their first contribution in https://github.com/oven-sh/bun/pull/5046
* @tomredman made their first contribution in https://github.com/oven-sh/bun/pull/4997
* @nazeelashraf made their first contribution in https://github.com/oven-sh/bun/pull/4990
* @jonathantneal made their first contribution in https://github.com/oven-sh/bun/pull/5146
* @Southpaw1496 made their first contribution in https://github.com/oven-sh/bun/pull/5143
* @MasterGordon made their first contribution in https://github.com/oven-sh/bun/pull/5120
* @cfal made their first contribution in https://github.com/oven-sh/bun/pull/4841
* @nathanhammond made their first contribution in https://github.com/oven-sh/bun/pull/5234
* @0x346e3730 made their first contribution in https://github.com/oven-sh/bun/pull/5227
* @amt8u made their first contribution in https://github.com/oven-sh/bun/pull/5061
* @zongzi531 made their first contribution in https://github.com/oven-sh/bun/pull/5243
* @G-Rath made their first contribution in https://github.com/oven-sh/bun/pull/5140
* @iidebyo made their first contribution in https://github.com/oven-sh/bun/pull/5057
* @gtramontina made their first contribution in https://github.com/oven-sh/bun/pull/5275
* @browner12 made their first contribution in https://github.com/oven-sh/bun/pull/5311
* @winghouchan made their first contribution in https://github.com/oven-sh/bun/pull/5294
* @philolo1 made their first contribution in https://github.com/oven-sh/bun/pull/5325
* @mroyme made their first contribution in https://github.com/oven-sh/bun/pull/5285
* @miccou made their first contribution in https://github.com/oven-sh/bun/pull/5225
* @desm made their first contribution in https://github.com/oven-sh/bun/pull/5115
* @tikotzky made their first contribution in https://github.com/oven-sh/bun/pull/5318
* @kryparnold made their first contribution in https://github.com/oven-sh/bun/pull/4905
* @TwanLuttik made their first contribution in https://github.com/oven-sh/bun/pull/4881
* @s0h311 made their first contribution in https://github.com/oven-sh/bun/pull/4452
* @chrisbodhi made their first contribution in https://github.com/oven-sh/bun/pull/5376
* @dylang made their first contribution in https://github.com/oven-sh/bun/pull/5379
* @DuGlaser made their first contribution in https://github.com/oven-sh/bun/pull/4647
* @nangchan made their first contribution in https://github.com/oven-sh/bun/pull/5419
* @sonyarianto made their first contribution in https://github.com/oven-sh/bun/pull/5425

**Full Changelog**: https://github.com/oven-sh/bun/compare/bun-v1.0.1...bun-v1.0.2

<hr /><em>This discussion was created from the release <a href='https://github.com/oven-sh/bun/releases/tag/bun-v1.0.2'>Bun v1.0.2</a>.</em>

**Comment by xieyuheng** [comment]
加油!加油!
